sdc/tests/test_dataframe.py

```diff
@@ -1663,6 +1664,26 @@ def test_impl():
         self.assertTrue(isinstance(two, np.ndarray))
         self.assertTrue(isinstance(three, np.ndarray))
 
+    def test_df_create_str_from_optional(self):
+        from sdc.hiframes.pd_dataframe_ext import get_dataframe_data
+        def test_impl():
+            df = pd.DataFrame({
+                        'A': ['a', 'b', None, 'a', '', 'None', 'b'],
+                        'B': ['a', 'b', 'd', 'a', '', 'c', 'b'],
+                        'C': [2, 1, np.nan, 1, np.nan, 2, 1]
+            })
+
+            total_elements = 0
+            column_ids = (0, 1, 2)
+            for col_id in literal_unroll(column_ids):
+                total_elements += len(get_dataframe_data(df, col_id))
+            return total_elements
+        hpat_func = self.jit(test_impl)
```

Comment:
You could decorate tested function directly:
```
@self.jit
def hpat_func():
    pass
```

Reply:
Fixed